src/equations/shallow_water_two_layer_2d.jl

```diff
@@ -0,0 +1,936 @@
+  h_upper_ll, _, _, h_lower_ll, _, _, _ = u_ll
+  h_upper_rr, _, _, h_lower_rr, _, _, _ = u_rr
+
+  # Extract and compute the velocities in the normal direction
+  v1_upper_ll, v2_upper_ll, v1_lower_ll, v2_lower_ll = velocity(u_ll, equations)
+  v1_upper_rr, v2_upper_rr, v1_lower_rr, v2_lower_rr = velocity(u_rr, equations)
+
+  v_upper_dot_n_ll = v1_upper_ll * normal_direction[1] + v2_upper_ll * normal_direction[2]
+  v_upper_dot_n_rr = v1_upper_rr * normal_direction[1] + v2_upper_rr * normal_direction[2]
+  v_lower_dot_n_ll = v1_lower_ll * normal_direction[1] + v2_lower_ll * normal_direction[2]
+  v_lower_dot_n_rr = v1_lower_rr * normal_direction[1] + v2_lower_rr * normal_direction[2]
+  
+  # Calculate averaged velocity of both layers
+  v_m_ll = (v_upper_dot_n_ll * h_upper_ll + v_lower_dot_n_ll * h_lower_ll) / (h_upper_ll + h_lower_ll)
+  v_m_rr = (v_upper_dot_n_rr * h_upper_rr + v_lower_dot_n_rr * h_lower_rr) / (h_upper_rr + h_lower_rr)
+
+  # Compute the wave celerity on the left and right
+  h_upper_ll, h_lower_ll = waterheight(u_ll, equations)
+  h_upper_rr, h_lower_rr = waterheight(u_rr, equations)
+
+  c_ll = sqrt(equations.gravity * (h_upper_ll + h_lower_ll))
+  c_rr = sqrt(equations.gravity * (h_upper_rr + h_lower_rr))
+
+  # The normal velocities are already scaled by the norm
+  return max(abs(v_m_ll), abs(v_m_rr)) + max(c_ll, c_rr) * norm(normal_direction)
+end
+
+
+# Specialized `DissipationLocalLaxFriedrichs` to avoid spurious dissipation in the bottom topography
+@inline function (dissipation::DissipationLocalLaxFriedrichs)(u_ll, u_rr, 
+    orientation_or_normal_direction, equations::ShallowWaterTwoLayerEquations2D)
+  λ = dissipation.max_abs_speed(u_ll, u_rr, orientation_or_normal_direction, equations)
+  diss = -0.5 * λ * (u_rr - u_ll)
+  return SVector(diss[1], diss[2], diss[3], diss[4], diss[5], diss[6], zero(eltype(u_ll)))
+end
+
+
+# Absolute speed of the barotropic mode
+@inline function max_abs_speeds(u, equations::ShallowWaterTwoLayerEquations2D)
+  h_upper, h_v1_upper, h_v2_upper, h_lower, h_v1_lower, h_v2_lower, _ = u
+
+  # Calculate averaged velocity of both layers
+  v1_m = (h_v1_upper + h_v1_lower) / (h_upper + h_lower)
+  v2_m = (h_v2_upper + h_v2_lower) / (h_upper + h_lower)
+
+  h_upper, h_lower = waterheight(u, equations)
+  v1_upper, v2_upper, v1_lower, v2_lower = velocity(u, equations)
+
+  c = sqrt(equations.gravity * (h_upper + h_lower)) 
+  return (max(abs(v1_m) + c, abs(v1_upper), abs(v1_lower)), 
+          max(abs(v2_m) + c, abs(v2_upper), abs(v2_lower)))
+end
+
+
+# Helper function to extract the velocity vector from the conservative variables
+@inline function velocity(u, equations::ShallowWaterTwoLayerEquations2D)
+  h_upper, h_v1_upper, h_v2_upper, h_lower, h_v1_lower, h_v2_lower, _ = u
+
+  v1_upper = h_v1_upper / h_upper
+  v2_upper = h_v2_upper / h_upper
+  v1_lower = h_v1_lower / h_lower
+  v2_lower = h_v2_lower / h_lower
+
+  return v1_upper, v2_upper, v1_lower, v2_lower
```

Comment:
```suggestion
  return SVector(v1_upper, v2_upper, v1_lower, v2_lower)
```
For consistency with other functions?

Reply:
Done in 4f2114ca248344248107f39fd2d7e0523083b079